src/Homie/Boot/BootConfig.cpp

```diff
@@ -278,99 +295,79 @@ void BootConfig::_onDeviceInfoRequest() {
-      jsonSetting["required"] = setting->isRequired();
-      if (!setting->isRequired()) {
-        jsonSetting["default"] = setting->get();
+
+    if (iSetting->type() != "unknown") {
+      jsonSetting["name"] = iSetting->getName();
+      jsonSetting["description"] = iSetting->getDescription();
+      jsonSetting["type"] = iSetting->type();
+      jsonSetting["required"] = iSetting->isRequired();
+
+      if (!iSetting->isRequired()) {
+        if (iSetting->isBool()) {
+          HomieSetting<bool>* setting = static_cast<HomieSetting<bool>*>(iSetting);
+          jsonSetting["default"] = setting->get();
+
+        }
+        else if (iSetting->isLong()) {
+          HomieSetting<long>* setting = static_cast<HomieSetting<long>*>(iSetting);
+          jsonSetting["default"] = setting->get();
+        }
+        else if (iSetting->isDouble()) {
+          HomieSetting<double>* setting = static_cast<HomieSetting<double>*>(iSetting);
+          jsonSetting["default"] = setting->get();
+        }
+        else if (iSetting->isConstChar()) {
+          HomieSetting<const char*>* setting = static_cast<HomieSetting<const char*>*>(iSetting);
+          jsonSetting["default"] = setting->get();
+        }
       }
     }
 
     settings.add(jsonSetting);
   }
 
-  size_t jsonBufferLength = json.measureLength() + 1;
-  std::unique_ptr<char[]> jsonString(new char[jsonBufferLength]);
-  json.printTo(jsonString.get(), jsonBufferLength);
-  _http.send(200, FPSTR(PROGMEM_CONFIG_APPLICATION_JSON), jsonString.get());
+  String output;
+  json.printTo(output);
+
+  request->send(200, FPSTR(PROGMEM_CONFIG_APPLICATION_JSON), output);
 }
 
-void BootConfig::_onNetworksRequest() {
+void BootConfig::_onNetworksRequest(AsyncWebServerRequest *request) {
   Interface::get().getLogger() << F("Received networks request") << endl;
   if (_wifiScanAvailable) {
-    _http.send(200, FPSTR(PROGMEM_CONFIG_APPLICATION_JSON), _jsonWifiNetworks);
-  } else {
-    _http.send(503, FPSTR(PROGMEM_CONFIG_APPLICATION_JSON), FPSTR(PROGMEM_CONFIG_NETWORKS_FAILURE));
+    request->send(200, FPSTR(PROGMEM_CONFIG_APPLICATION_JSON), _jsonWifiNetworks);
+  }
+  else {
+    _SendJSONError(request, F("Initial Wi-Fi scan not finished yet"), 503);
   }
 }
 
-void BootConfig::_onConfigRequest() {
+void BootConfig::_onConfigRequest(AsyncWebServerRequest *request) {
   Interface::get().getLogger() << F("Received config request") << endl;
   if (_flaggedForReboot) {
-    Interface::get().getLogger() << F("✖ Device already configured") << endl;
-    String errorJson = String(FPSTR(PROGMEM_CONFIG_JSON_FAILURE_BEGINNING));
-    errorJson.concat(F("Device already configured\"}"));
-    _http.send(403, FPSTR(PROGMEM_CONFIG_APPLICATION_JSON), errorJson);
+    _SendJSONError(request, F("✖ Device already configured"), 403);
     return;
   }
 
-  StaticJsonBuffer<MAX_JSON_CONFIG_ARDUINOJSON_BUFFER_SIZE> parseJsonBuffer;
-  std::unique_ptr<char[]> bodyString = Helpers::cloneString(_http.arg("plain"));
```

Comment:
What's wrong with `unique_ptr<>`?

Reply:
TBH not the best a c++, dont know all its ins and outs, but [_tempObject](https://github.com/me-no-dev/ESPAsyncWebServer/blob/8604f67a5ff3d48ddc725a8606dcef774d9d5567/src/WebRequest.cpp#L91) is freed by the request when done. Thus is there need for this? Id love to know best practices.

Reply:
No, indeed, you're right. Didn't know how ESPAsyncWebServer worked.